app/views/pages/developer-guide/limitations.liquid

```diff
@@ -0,0 +1,112 @@
+
+The most important limitation when it comes to Page is the execution time - we will throw 504 Timeout error after 9 seconds (configurable in a private stack). Typically issue hitting this limitation is not leveraging [Background Jobs](/best-practices/backend-performance/background-jobs) for potentially time-consuming and non-critical operations, like sending emails and API calls to a third-party system, etc.
+
+To protect us from Denial-of-Service (DoS) attacks, our Web Application Firewall will block any IP that submits more than 300 requests within 5 5-minute period for about 10 minutes.
+
+### Partials
+
+We expect you to have up to tens of thousands of partials.
+
+### Translations
+
+We expect you to have a few thousand translations per language and tens of languages.
+
+## pos-cli deploy
+
+There is no hard limit in terms of how many files can be deployed to platformOS.
+ You can upload gigabytes of assets (although in such cases, there is a high chance that you should be using Uploads instead of Assets).
+There is a default soft limit of 90MB for the release.zip, which pos-cli generates with all of the configuration files - pages, partials, forms, translations, etc. (everything except assets).
+
+## instance copy / data import / data export
+
+Instance copy is a feature in Partner Portal that allows you to quickly move data and uploads from one instance to another, effectively creating almost a perfect copy (only background jobs are not migrated). We expect instance copy to work correctly for up to a couple of hundreds of thousands of records and gigabytes of uploads - the more records and data, the longer it will take.
+
+Instance copy relies behind the scenes on data export and data import features, which have the same expectations.
+However, there is a hard limit of 90MB file size of the deployment for configuration files (everything except of Assets).
+
+## Session
+
+You can store values per session using [session](/api-reference/liquid/platformos-tags#session) tag. You should be mindful that sessions are stored in Redis. To maintain efficiency, it is best to keep payloads as small as possible and store only essential data in sessions. If you need to handle large data sets, consider storing the data in a database or file storage and passing a reference (such as an ID or file URL). One of the most common example in a context of eCommerce is to store in the session the id of current user shopping cart as a `cart_id`, instead of an object with names of the products, prices etc.
```

Comment:
```suggestion
You can store values per session using [session](/api-reference/liquid/platformos-tags#session) tag. It's important to note that sessions are stored in Redis. To optimize efficiency, it's advisable to keep payloads as small as possible and store only essential data in sessions. If you need to handle large datasets, consider storing the data in a database or file storage and passing a reference (such as an ID or file URL). A common example in the context of eCommerce is storing the ID of the current user's shopping cart as `cart_id` in the session, rather than storing an object with names of the products, prices, etc.
```